python_modules/libraries/dagster-gcp/dagster_gcp/dataproc/resources.py

```diff
@@ -149,9 +155,75 @@ def cluster_context_manager(self):
             self.delete_cluster()
 
 
+class DataprocResource(ConfigurableResource):
+    project_id: str = Field(
+        description=(
+            "Required. Project ID for the project which the client acts on behalf of. Will be"
+            " passed when creating a dataset/job."
+        )
+    )
+    region: str = Field(description="The GCP region.")
+    cluster_name: str = Field(
+        description=(
+            "Required. The cluster name. Cluster names within a project must be unique. Names of"
+            " deleted clusters can be reused."
+        )
+    )
+    cluster_config_yaml_path: Optional[str] = Field(
+        default=None,
+        description=(
+            "Full path to a YAML file containing cluster configuration. See"
+            " https://cloud.google.com/dataproc/docs/reference/rest/v1/ClusterConfig for"
+            " configuration options."
+        ),
+    )
+    cluster_config_json_path: Optional[str] = Field(
+        default=None,
+        description=(
+            "Full path to a JSON file containing cluster configuration. See"
+            " https://cloud.google.com/dataproc/docs/reference/rest/v1/ClusterConfig for"
+            " configuration options."
+        ),
+    )
+    cluster_config_dict: Optional[str] = Field(
+        default=None,
+        description=(
+            "Python dictionary containing cluster configuration. See"
+            " https://cloud.google.com/dataproc/docs/reference/rest/v1/ClusterConfig for"
+            " configuration options."
+        ),
+    )
+
+    def _read_yaml_config(self, path: str) -> Mapping[str, Any]:
+        with open(path, "r", encoding="utf8") as f:
+            return yaml.safe_load(f)
+
+    def _read_json_config(self, path: str) -> Mapping[str, Any]:
+        with open(path, "r", encoding="utf8") as f:
+            return json.load(f)
+
+    def get_client(self) -> DataprocClient:
+        cluster_config = None
+        if self.cluster_config_json_path:
```

Comment:
nit: raise an error if more than one config type is specified, rather than silently preferring one